packages/vm/src/compatability.rs

```diff
@@ -40,11 +42,46 @@ pub fn check_wasm(wasm_code: &[u8]) -> Result<()> {
             ));
         }
     };
+    check_wasm_memories(&module)?;
     check_wasm_exports(&module)?;
     check_wasm_imports(&module)?;
     Ok(())
 }
 
+fn check_wasm_memories(module: &Module) -> Result<()> {
+    let section = match module.memory_section() {
+        Some(section) => section,
+        None => {
+            return make_validation_err("Wasm contract doesn't have a memory section".to_string());
+        }
+    };
+
+    let memories = section.entries();
+    if memories.len() != 1 {
+        return make_validation_err("Wasm contract must contain exactly one memory".to_string());
+    }
+
+    for memory in memories {
```

Comment:
Why not just use `let memory = memories[0]` since we know this is set? And avoid the for loop

Reply:
What you suggest is perfectly correct. I'm just scared to use the non-typesafe [] operator because the length check is 3 lines away and when the above check gets refactored, the rest of the code breaks. Which is crazy paranoid. Will update.